packages/block-library/src/post-comments-form/view.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import './runtime/init.js';
+import { store, navigate } from './runtime';
+
+store( {
+	state: {
+		core: {
+			commentsFormError: '',
+		},
+	},
+	actions: {
+		core: {
+			commentsFormSubmission: async ( { event, state } ) => {
+				event.preventDefault();
+
+				state.core.commentsFormError = '';
+
+				const formData = new FormData( event.target );
+
+				const res = await fetch(
+					'http://localhost:8888/wp-comments-post.php',
+					{
+						method: 'POST',
+						body: formData,
+					}
+				);
+
+				const html = await res.text();
+
+				if ( res.status !== 200 ) {
+					const dom = new window.DOMParser().parseFromString(
+						html,
+						'text/html'
+					);
+					state.core.commentsFormError =
+						dom.querySelector( 'p' ).innerHTML;
+				} else {
+					navigate( res.url, {
```

Comment:
Normally when leaving a comment, the submission will take you to the permalink for the comment. For example, `https://localhost:8888/?p=1#comment-2`. But here it seems `navigate` doesn't include the URL fragment. I think this should be retained as it is normal WP behavior and it facilitates linking/bookmarking. With that URL fragment in place, it would also be a nice touch if there was a yellow fade highlight applied to the comment via the `.comment:target` CSS selector to make it abundantly clear where there newly-added comment is showing up.

Reply:
> But here it seems navigate doesn't include the URL fragment. I think this should be retained as it is normal WP behavior and it facilitates linking/bookmarking.

The client router doesn't support hashes yet, but I've opened an issue to track it. Thanks 🙂

- https://github.com/WordPress/block-interactivity-experiments/issues/202

> it would also be a nice touch if there was a yellow fade highlight applied to the comment via the .comment:target CSS selector to make it abundantly clear where there newly-added comment is showing up

That would be a very nice touch indeed.